libcurv/output_file.cc

```diff
@@ -6,27 +6,49 @@
 #include <libcurv/context.h>
 #include <libcurv/exception.h>
 extern "C" {
-#include <sys/types.h>
-#include <sys/stat.h>
 #include <fcntl.h>
-#include <stdio.h>
 #include <time.h>
 #include <unistd.h>
+
+#ifdef _WIN32
+    #include <Windows.h>
+#else
+    #include <sys/types.h>
+    #include <sys/stat.h>
+
+#endif
 }
 #include <fstream>
+#include <string>
 
 namespace curv {
 
 namespace fs = Filesystem;
 namespace io = boost::iostreams;
 
+#ifdef _WIN32
+    // Format an error code from the Windows API (e.g. as obtained via GetLastError) as a std::wstring
+    //
+    // Original Source: <https://stackoverflow.com/a/17386923>
+    // Author: LeviX <https://stackoverflow.com/users/747145/levix>
+    // License: CC BY-SA 3.0
+    std::wstring win_strerror(const DWORD errorCode) {
+      // TODO: Is this idiomatic C++ code?
+      wchar_t buf[256];
+      FormatMessageW(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                   NULL, GetLastError(), MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), 
+                   buf, (sizeof(buf) / sizeof(wchar_t)), NULL);
+      return std::wstring(buf);
+    }
+#endif
+
```

Comment:
@doug-moen I wasn't sure where to put that function. It could be reused by other code for formatting error messages by WinAPI.